Cange the Icon path accordingly.
We place a synergy.xpm /usr/share/pixmaps
Index: synergy-1.4.5/res/synergy.desktop
===================================================================
--- synergy-1.4.5.orig/res/synergy.desktop	2011-10-30 03:57:41.000000000 +0100
+++ synergy-1.4.5/res/synergy.desktop	2011-11-11 19:22:04.280083769 +0100
@@ -2,6 +2,6 @@
 Name=Synergy
 Comment=Share your keyboard and mouse over a network
 Exec=synergy
-Icon=/usr/share/icons/synergy.ico
+Icon=synergy
 Type=Application
 Categories=Utility
